beaker/application.py

```diff
@@ -1,418 +1,963 @@
-import base64
-from inspect import getattr_static
-from typing import Final, Any, cast, Optional
-from algosdk.v2client.algod import AlgodClient
-from algosdk.abi import Method
-from pyteal.compiler.compiler import FRAME_POINTERS_VERSION
+import dataclasses
+import inspect
+import typing
+import warnings
+from collections.abc import Callable, Iterator, MutableMapping
+from contextlib import contextmanager
+from contextvars import ContextVar
+from typing import (
+    Concatenate,
+    Generic,
+    Literal,
+    ParamSpec,
+    TypeAlias,
+    TypeVar,
+    cast,
+    overload,
+)
+
 from pyteal import (
-    SubroutineFnWrapper,
-    TealInputError,
-    Txn,
-    MAX_TEAL_VERSION,
     ABIReturnSubroutine,
+    Approve,
     BareCallActions,
+    Bytes,
+    CallConfig,
     Expr,
-    Global,
+    Int,
+    MethodConfig,
     OnCompleteAction,
-    OptimizeOptions,
     Router,
-    Bytes,
-    Approve,
+    SubroutineFnWrapper,
+    TealType,
+    Txn,
 )
 
-from beaker.decorators import (
-    get_handler_config,
+from beaker.application_specification import (
+    ApplicationSpecification,
+    DefaultArgumentDict,
     MethodHints,
-    MethodConfig,
-    create,
-    HandlerFunc,
 )
-
-from beaker.state import (
-    AccountState,
-    AccountStateBlob,
-    ApplicationStateBlob,
-    ApplicationState,
-    ReservedAccountStateValue,
-    AccountStateValue,
-    ApplicationStateValue,
-    ReservedApplicationStateValue,
-    prefix_key_gen,
+from beaker.build_options import BuildOptions
+from beaker.decorators import authorize as authorize_decorator
+from beaker.logic_signature import LogicSignature, LogicSignatureTemplate
+from beaker.precompile import (
+    PrecompiledApplication,
+    PrecompiledLogicSignature,
+    PrecompiledLogicSignatureTemplate,
 )
-from beaker.errors import BareOverwriteError
-from beaker.precompile import AppPrecompile, LSigPrecompile
-from beaker.lib.storage import List
+from beaker.state._aggregate import GlobalStateAggregate, LocalStateAggregate
+from beaker.state.primitive import GlobalStateValue, LocalStateValue
+
+if typing.TYPE_CHECKING:
+    from algosdk.v2client.algod import AlgodClient
+
+__all__ = [
+    "Application",
+    "this_app",
+    "precompiled",
+]
+
+OnCompleteActionName: TypeAlias = Literal[
+    "no_op",
+    "opt_in",
+    "close_out",
+    "update_application",
+    "delete_application",
+]
```

Comment:
missing clear state?

Reply:
yeah it was in there but removed when PyTeal introduced special handling. Good callout though, I found an omission where it should handle both these values or "clear_state" for consistency.